Feat/login

BUIHUUHAIDANG opened the pull request.

BUIHUUHAIDANG committed:
Merge pull request #1 from BUIHUUHAIDANG/dev

add coding convention

BUIHUUHAIDANG committed:
Add UserAccount.h: Define class structure and methods

BUIHUUHAIDANG committed:
update UserAccount

BUIHUUHAIDANG committed:
Delete src/Project1/UserAccount.h

BUIHUUHAIDANG committed:
Delete src/wallet-management directory

BUIHUUHAIDANG committed:
cấu trúc dữ liệu (lớp) quản lý thông tin tài khoản người dùng.

BUIHUUHAIDANG committed:
update

BUIHUUHAIDANG committed:
Merge branch 'feat/login' of github.com:BUIHUUHAIDANG/wallet-management into feat/login

BUIHUUHAIDANG committed:
fix UserAccount

BUIHUUHAIDANG committed:
fakehash funtion

BUIHUUHAIDANG committed:
fakehash header

BUIHUUHAIDANG committed:
Update create user from input and auto random password

BUIHUUHAIDANG committed:
update password in file and save userinfo

BUIHUUHAIDANG committed:
change fontend to english and add check login when sign up and when first login. Update main.cpp

BUIHUUHAIDANG committed:
change to english word

BUIHUUHAIDANG committed:
hello world

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+UserAccount::UserAccount(const string& uname, const string& name, const string& phone, bool manager) {
+	username = uname;
+	fullname = name;
+	phonenumber = phone;
```
camelCase

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+	fullname = name;
+	phonenumber = phone;
+	isManager = manager;
+	firstlogin = true;
```
camelCase

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+	phonenumber = phone;
+	isManager = manager;
+	firstlogin = true;
+	WalletID = "WalletID_" + uname;
```
camelCase

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+}
+void saveUsertofile(const UserAccount& user, const string& filename) {
+	ofstream file(filename, ios::app);
+	file << user.getUsername() << "," << user.getFullname() << "," << user.getPhonenumber() << "," << user.getPasswordHash() << "," << user.getWalletID() << "," << user.getisManager() << "," << user.getFirstlogin() << endl;
```
duplicate code

aidenSepiol replied:
write method private named serialize() -> return string (reuse that method on line 97)

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+	tempFile.close();
+
+	remove(filename.c_str());
+	rename("temp.txt", filename.c_str());
```
file name should be used as constant as  you need to use on others class as well

BUIHUUHAIDANG committed:
fix bug at UserAccount.cpp

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+	return updated;
+}
+bool loginAndHandleFirstLogin(const string& username, const string& password) {
+	ifstream inFile("users.txt");
```
put it in constant

aidenSepiol commented on src/login/login/UserAccount.cpp:
```diff
@@ -0,0 +1,176 @@
+	string line;
+	bool updated = false;
+
+	while (getline(inFile, line)) {
```
when does it break? is it scanning all lines, isn't that right?

Vuxvenn committed:
Sửa lỗi đăng nhập thất bại

Vuxvenn committed:
Merge branch 'main' of https://github.com/BUIHUUHAIDANG/wallet-management

Vuxvenn committed:
Bỏ Test git.cbp khỏi GitHub

Vuxvenn committed:
Xóa nhiều file không cần thiết

Vuxvenn committed:
Bỏ theo dõi thư mục bin/Debug

Vuxvenn committed:
xóa file dư

Vuxvenn committed:
Add main.cpp

Vuxvenn committed:
Merge branch 'feat/login' of https://github.com/BUIHUUHAIDANG/wallet-management into feat/login

Vuxvenn committed:
Add files via upload

Sửa lỗi đăng nhập không thành công

Vuxvenn committed:
Sửa lỗi đăng nhập không thành công

Vuxvenn committed:
fixbug

Vuxvenn committed:
Sửa lỗi đăng nhập không thành công

BUIHUUHAIDANG committed:
update

BUIHUUHAIDANG committed:
Merge branch 'feat/login' of github.com:BUIHUUHAIDANG/wallet-management into feat/login

BUIHUUHAIDANG committed:
update

Vuxvenn committed:
Giờ đây có thể tạo nhiều user

Vuxvenn committed:
Giờ đây không thể đăng ký username trùng nhau

BUIHUUHAIDANG committed:
update duplicate header

BUIHUUHAIDANG committed:
Merge branch 'feat/login' of github.com:BUIHUUHAIDANG/wallet-management into feat/login

BUIHUUHAIDANG committed:
update duplicate header

BUIHUUHAIDANG committed:
update duplicate funtion

BUIHUUHAIDANG committed:
update main

BUIHUUHAIDANG committed:
fix bug updatepasswordinfile

imagine363 committed:
Add files via upload

admin and user panel 0: user 1: admin chua co otp

Vuxvenn committed:
Chèn phần của Minh vào main

Vuxvenn committed:
Admin có thể thay đổi username của 1 user

BUIHUUHAIDANG committed:
fix admin menu and mix code

BUIHUUHAIDANG committed:
Merge branch 'feat/login' of github.com:BUIHUUHAIDANG/wallet-management into feat/login

BUIHUUHAIDANG committed:
change to english

BUIHUUHAIDANG committed:
change to english

BUIHUUHAIDANG committed:
git push wallet header and struct transation

BUIHUUHAIDANG committed:
add constuctor Wallet and Transation

imagine363 committed:
user and admin panels now have OTP functions

BUIHUUHAIDANG committed:
update useredit and update MenuAdmin

BUIHUUHAIDANG committed:
Merge branch 'feat/login' of github.com:BUIHUUHAIDANG/wallet-management into feat/login

BUIHUUHAIDANG committed:
update edit users and update menuadmin

BUIHUUHAIDANG committed:
fix camel

MasterNgPhat committed:
File otp.cpp 

OTP funtion

MasterNgPhat committed:
include<otp.h>

libary otp function

imagine363 committed:
otpmanager

imagine363 committed:
updated main

imagine363 committed:
user and admin

BUIHUUHAIDANG committed:
update

imagine363 committed:
aaa

imagine363 committed:
222

imagine363 committed:
3333

BUIHUUHAIDANG committed:
Update

BUIHUUHAIDANG committed:
update

BUIHUUHAIDANG committed:
update

BUIHUUHAIDANG committed:
update cpp

Vuxvenn committed:
Update tính năng trong admin user

MasterNgPhat committed:
all function main for otp and int main.cpp

MasterNgPhat committed:
header file for otp.cpp (function)

imagine363 committed:
cua t

imagine363 committed:
Update cua minh

imagine363 committed:
Update cua minh

imagine363 committed:
functions

BUIHUUHAIDANG committed:
update today

BUIHUUHAIDANG committed:
update today

Vuxvenn committed:
Khởi tạo ví cho user

- Thêm chức năng wallet vào user menu
- Trong wallet có: xem thông tin ví và giao dịch
- Thông tin ví có: Tên ví, Số dư và lịch sự giao dịch
[x] Chưa hoàn thiện chức năng chuyển tiền (giao dịch)

Vuxvenn committed:
Sửa lại hàm class wallet và tạo ví cho user, chưa có chức năng chuyển tiền

Vuxvenn committed:
[BIG UPDATE] Tối ưu cách khởi tạo ví, thêm chức năng chuyển tiền thông qua username

Sửa thuật toán khởi tạo ví và chuyển tiền, sửa cách lấy thông tin trong file liên quan đến ví

Vuxvenn committed:
Fix bug and update

Vuxvenn committed:
Xóa function dư
